CMakeLists.txt

```diff
@@ -7402,10 +7403,14 @@ IF(MSVC)
   SET_PROPERTY(SOURCE ${ALL_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O2 >")
   SET_PROPERTY(SOURCE ${HOT_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O2 >")
   SET_PROPERTY(SOURCE ${COLD_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: /O1 >")
-ELSE()
-  SET_PROPERTY(SOURCE ${ALL_MICROKERNEL_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: -O2 >")
-  SET_PROPERTY(SOURCE ${HOT_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: -O2 >")
-  SET_PROPERTY(SOURCE ${COLD_SRCS} APPEND_STRING PROPERTY COMPILE_FLAGS "$<$<NOT:$<CONFIG:Debug>>: -Os >")
+ELSEIF(NOT (CMAKE_BUILD_TYPE AND (CMAKE_BUILD_TYPE STREQUAL "Debug")))
+  # Compataible with Xcode, which does not support per-config per-source settings,
```

Comment:
We can't set `"$<$<NOT:$<CONFIG:Debug>>: -O2 >"` `COMPILE_FLAGS` on source files with XCode, but we can do it on targets. So you need to set `COMPILE_FLAGS` for targets rather than source files, then it will work with XCode. Trace `ALL_MICROKERNELS_SRCS`, `HOT_SRCS`, `COLD_SRCS` to targets and set `COMPILE_FLAGS` for those targets accordingly.

Reply:
TARGET_COMPILE_OPTIONS seems work. but not quite sure it 's the best way to do that.